src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -177,7 +178,16 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
                     }
                     const resources = getResourcesFromTemplateDatum(templateDatum)
                     for (const resourceKey of resources.keys()) {
-                        const runtimeName = resources.get(resourceKey)?.Properties?.Runtime
+                        const resource = resources.get(resourceKey)
+                        // Skip it if there is not actually a resource or it's not a funciton
```

Comment:
maybe this should be done in `getResourcesFromTemplateDatum()`, so then `resources` only contains what we want and we don't need a `continue` here.

`getResourcesFromTemplateDatum()` could take an optional `resourceTypes` list which filters its result.

Reply:
I think an optional resourceTypes works, will add